tiingo/wsclient.py

```diff
@@ -38,7 +38,7 @@ def cb_fn(msg):
     while True:pass
     '''
 
-    def __init__(self,config={},endpoint=None,on_msg_cb=None):
+    def __init__(self,config,endpoint=None,on_msg_cb=None):
```

Comment:
Thanks for updating this line, to complete the change we should also add the following ternary

```
self.config = {} if config is None else config
```

Otherwise, we'll get an error when we try to run `self.config.update` as `update` isn't available on a `None` type.

Reply:
I'm not sure why the travis checks aren't showing in the Github UI, but you can check the test results from the Travis ui here:

https://travis-ci.org/github/hydrosquall/tiingo-python/jobs/733171252

Reply:
but like. that should only happen if someone called ```TiingowebsocketClient``` without the config param. python would raise an error. because the param doesn't have a default argument. so it won't even reach the ```self.config.update```.
am i getting this right?

Reply:
Ah, you're right - I have been working mostly in JS recently, and forgot that in Python, if you don't pass in an argument that you'll get an error at runtime (in JS, if you don't pass in an argument, the function runs but assumes that the value of config === `undefined`. I supposed the value of doing the check would have been to guard against someone doing `TiingoWebsocketClient(config=None)`.

Anyways, I think that having a named argument is helpful to callers so they know the meaning of the position, so I guess the full change would look like:

```python
def __init__(config=None, # other args)

    self.config = {} if config is None else config
```

Regarding the failing test, it appears that the error wasn't raised because the test environment contained the TIINGO_API_KEY environment variable. In order to complete the change, you can follow the `delenv` example here:

```
monkeypatch.delenv
```

https://docs.pytest.org/en/4.6.1/monkeypatch.html#monkeypatching-environment-variables

In case you haven't run the unit tests locally before, that can be done by running the following in your terminal when in the root directory, this will be faster than waiting for the tests to run in CI. 

```bash
python setup.py test
```

Reply:
Hi @mohamedemad4, just wanted to check if you might have time to take a look at fixing this test - I think it's the last blocking item that we'd need before we can include your changes in the new release.

Reply:
the link you provided is broken

Reply:
also i don't get why the CI tests are failing? shouldn't the CI be providing it's own TINGO_API_KEY as an env var? or am i missing something?

Reply:
Hi @Mohamededmad4, thanks for looking into this. The CI environment does have a `TIINGO_API_KEY`. The test is failing because the `TIINGO_API_KEY` is provided, which is why `test_missing_api_key` doesn't work.

https://github.com/hydrosquall/tiingo-python/blob/f6ef473b0d8e951ede092509f17854f7e0ffd053/.github/workflows/python-package.yml#L42-L43

As a result, the code uses that environment variable and we never get to test the "RuntimeError" check for when the config is empty AND nothing is specified in the environment variables. To fix it, you can either change the logic of the client code, or delete the environment variable just inside of that test.

https://github.com/hydrosquall/tiingo-python/blob/be25aa24c3dc2551de164e9ef8bb03a7a30c1f93/tiingo/wsclient.py#L46-L56

Let me know if that's still confusing. The updated link for how to patch environment variables is here:

https://docs.pytest.org/en/stable/monkeypatch.html#monkeypatching-environment-variables